### PR TITLE
Add Status method to Conn interface

### PR DESCRIPTION
**Pull request by tylertreat**
Expose the underlying NATS conn status. Addresses #67.

### CODE REVIEW

File: stan.go [hunk 1]
```diff
@@ -50,6 +50,9 @@ type Conn interface {
 
 	// Close
 	Close() error
+
+	// Status returns the status of the underlying NATS conn.
+	Status() nats.Status
```
Comment (ColinSullivan1) [inline stan.go]: What do you think of an API that returns the underlying NATS conn itself (versus additional wrappers around NATS conn APIs)?  cc: @kozlovic @aricart

Comment (aricart) [inline stan.go]: I ran into exactly the same thing. I think returning the nats connection is best. I was looking for the Statistics, and had to resort to create my own nats connection and create my streaming connection with it. With the connection we get access to the whole thing.

**Comment by kozlovic** [comment]
Given that one can pass a NATS connection as an option, with all the pros but also cons that were discussed, I also think that returning the underlying NATS Connection would be more flexible. @tylertreat , would you agree?
If so, would you want to change this PR to return the connection instead? We can certainly do that ourselves, but wanted to give you the opportunity to contribute. Let us know.
Thanks!

**Comment by tylertreat** [comment]
Yep, I will make those changes.

**Comment by aricart** [comment]
One thing to keep in mind is that the callbacks and options are not going to work. So the client code could break streaming... I think the correct thing is to expose Status and Statistics that would be *safer*

**Pull request closed without merging**